mscore/metaedit.cpp

```diff
@@ -29,95 +30,263 @@ namespace Ms {
 //---------------------------------------------------------
 
 MetaEditDialog::MetaEditDialog(Score* s, QWidget* parent)
-   : QDialog(parent)
+   : QDialog(parent),
+     m_score(s),
+     m_dirty(false)
       {
-      setObjectName("MetaEditDialog");
+      if (objectName().isEmpty())
+            setObjectName("MetaEditDialog");
+
       setupUi(this);
-      setWindowFlags(this->windowFlags() & ~Qt::WindowContextHelpButtonHint);
-      score = s;
-      dirty = false;
+      QDialog::setWindowFlag(Qt::WindowContextHelpButtonHint, false);
+      QDialog::setWindowFlag(Qt::WindowMinMaxButtonsHint);
+
+      QString mscoreVrs = m_score->mscoreVersion();
+      if (!mscoreVrs.isEmpty())
+            version->setText(mscoreVrs);
 
-      level->setValue(score->mscVersion());
-      version->setText(score->mscoreVersion());
-      int rev = score->mscoreRevision();
+      int scoreVrs = m_score->mscVersion();
+      if (scoreVrs)
+            level->setText(QString::number(scoreVrs, 10)); // in decimal
+
+      int rev = m_score->mscoreRevision();
       if (rev > 99999)  // MuseScore 1.3 is decimal 5702, 2.0 and later uses a 7-digit hex SHA
             revision->setText(QString::number(rev, 16));
       else
             revision->setText(QString::number(rev, 10));
-      filePath->setText(score->importedFilePath());
-      filePath->setTextInteractionFlags(Qt::TextSelectableByMouse);
 
-      int idx = 0;
+      QString sfilePath(m_score->importedFilePath());
+      if (!sfilePath.isEmpty())
+            filePath->setText(m_score->importedFilePath());
+
       QMapIterator<QString, QString> i(s->metaTags());
-      QGridLayout* grid = static_cast<QGridLayout*>(scrollWidget->layout());
       while (i.hasNext()) {
             i.next();
-            QLabel* l = new QLabel;
-            l->setText(i.key());
-            QLineEdit* text = new QLineEdit(i.value(), 0);
-            connect(text, SIGNAL(textChanged(const QString&)), SLOT(setDirty()));
-            grid->addWidget(l, idx, 0);
-            grid->addWidget(text, idx, 1);
-            ++idx;
+
+            QString s = i.key();
+            if (isSystemTag(s)) {
+                  QLineEdit *tag = new QLineEdit;
+                  tag->setStyleSheet("QLineEdit { background: transparent; }"); // Couldn't find any other way to remove the background.
+                  tag->setFrame(false);
+                  tag->setFocusPolicy(Qt::NoFocus);
+                  tag->setReadOnly(true);
+                  tag->setToolTip(tr("This is a system tag. It's name can't be modified."));
```

Comment:
This is a system tag. **Its** name can't be modified
(its = possessive; it's = "it is" contracted.)

Reply:
oops, sorry for that one.